### ISSUE TITLE
Error downloading NIFTI

### ISSUE THREAD

**Issue by mgaubert** [fli-iam/shanoir-ng#1447]
Hello,

I try to download some data acquired in 2018 from the protocol EMISEP in Nifti format. Using shanoir_downloader, it is working fine but for one image that could not be downloaded (maybe an issue with a space or something like this). 
And when I try to download this image using Solr, I have an error:

`08:52 - Error from datasets server

    2022-11-07 08:52:33
    [422] UNPROCESSABLE_ENTITY
    https://shanoir.irisa.fr/shanoir-ng/datasets/datasets/massiveDownload
    unknown cause
`

When downloading in DICOM format, it is working fine. But the conversion should be done with the same tool (and version) as for the other images, dcm2nii_2014-08-04, and I can't find this version anymore. 

Malo

**Comment by jcomedouteau** [comment]
Bonjour Malo,

En effet, il n'y a pas de fichier nifti associé à ce dataset, je vais essayer de comprendre pourquoi il n'y a pas eu de conversion.
Tu peux reconvertir ce dataset depuis le détail du dataset: https://shanoir-qualif.irisa.fr/shanoir-ng/dataset/details/275830 en utilisant le bouton "convert nifti", tu pourras alors le re-télécharger en utilisant le convertisseur de ton choix (dcm2nii_2014-08-04 est bien disponible normalement).
Merci pour ton retour et bonne journée !

Jean-Côme

**Comment by mgaubert** [comment]
Salut Jean-Côme,
OK, merci pour l'astuce, je n'avais pas pensé à tester comme ça.
Néanmoins, je suis intéressé pour une autre fois si tu trouves la solution au problème. 
Bonne journée à toi également,
Malo

**Comment by michaelkain** [comment]
Dear Malo, thank you for your issue. EMISEP is one of the pretty old studies in Shanoir, that has been imported using shanoir-old. It is quite impossible for us to go back into the code 5-8 years ago. Repo deleted. I might have a copy but to relate which version it was, I would not know how to do it. As long as the new version works, we might stop here. I will close the issue, if fine for you?

**Comment by mgaubert** [comment]
The DICOM are available. Could it be possible to redo the conversion to NIFTI directly on the server and thus avoid to go through each individual image details?

**Comment by michaelkain** [comment]
From what you wrote about, I thought it might be a rare phenomenon, that can be manually solved? using the details. If it is a mass phenomenon in your study, we will have to develop a new feature to re-apply on all existing datasets, you might select in the search. The question might be, do we want to keep the ancient conversions or do we in parallel store all reconverted. So, not an easy task and not on the roadmap until 2023. But we keep it in mind. We discussed that some times with Elise as well, but the solution is far from trivial, e.g. as we might just double the storage space for each converter, which requires a new referencing. Maybe, we could add a feature during the download moment, per dataset, re-convert with a given converter and propose that for download. The zip created for download will then be erased after download. With kind regards, Michael

**Comment by mgaubert** [comment]
Actually, I work from data from 2018. And here, it happens sometimes but not very often. So it is manageable to do it manually so far, especially because I am using `shanoir_downloader` of Arthur + modification by Quentin and myself and it does not seem to be a problem with this tool.
But for some images, `shanoir_downloader` can't find the data (a problem of space or tabulation in the name or... I don't know why) and it may be more annoying when we will download older data (before 2018) because in this case, I have the feeling that the problem will happen more frequently.
Then, I understand that it is quite demanding in term of human and computer ressources and that it is not on your priority list at the moment. So maybe, we will try to download the DICOM (no problem for this data usually) and do the conversion afterwards... I have to think about it and talk with others. Or the feature you are talking about at the end of your message could do the job I guess.

**Comment by michaelkain** [comment]
Hi, I will close that issue, as there is a new issue for the roadmap to propose Mass NIfTI conversion via the frontend, that should solve this issue.
